src/todo-config.ts

```diff
@@ -94,33 +101,60 @@ export function getTodoConfig(
   return todoConfig;
 }
 
-function getFromConfigFile(basePath: string, engine: string): TodoConfig | undefined {
-  const pkg = requireFile(basePath, 'package.json');
-  const lintTodorc = requireFile(basePath, '.lint-todorc.js');
+/**
+ * Validates whether we have a unique config in a single location.
+ *
+ * @param baseDir - The base directory that contains the project's package.json.
+ * @returns A ConfigValidationResult that indicates whether a config is unique
+ */
+export function validateConfig(baseDir: string): ConfigValidationResult {
+  const pkg = requireFile(baseDir, 'package.json');
+  const lintTodorc = requireFile(baseDir, '.lint-todorc.js');
+  const validationResult: ConfigValidationResult = {
+    pkg,
+    lintTodorc,
+    isValid: !(pkg?.lintTodo && lintTodorc),
+  };
+
+  if (!validationResult.isValid) {
+    validationResult.message =
+      'You cannot have todo configurations in both package.json and .lint-todorc.js. Please move the configuration from the package.json to the .lint-todorc.js';
+  }
+
+  return validationResult;
+}
 
-  if (pkg?.lintTodo && lintTodorc) {
-    throw new Error(
-      'You cannot have todo configurations in both package.json and .lint-todorc.js. Please move the configuration from the package.json to the .lint-todorc.js'
-    );
+function isTodoConfig(config: TodoConfig | TodoConfigByEngine): config is TodoConfig {
+  return (
+    Object.keys(config).length === 0 || Object.prototype.hasOwnProperty.call(config, 'daysToDecay')
+  );
+}
```

Comment:
FYI - for others, this lets us narrow from `TodoConfig | TodoConfigByEngine` to just `TodoConfig`.

https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates